src/filesystem.cpp

```diff
@@ -0,0 +1,189 @@
+#include <Windows.h>
```

Comment:
is including `Windows.h` going to break compiles on linux?
I assume this is because you're hooking a windows compiled `.dll` for the vpk parsing